src/js/parse-content.js

```diff
@@ -46,6 +46,31 @@ module.exports = function parseContent (options) {
 
     return obj
   }
+  /**
+   * Convert heading elements to plain objects
+   * @param {Array} headingsArray
+   * @return {Array}
+   */
+  function getAllHeadingsObject (headings) {
+    var headingsObject = Array.from(headings).map(function (heading) {
+      return getHeadingObject(heading)
+    })
+
+    if (options.headingsLabelCallback) {
+      var allLabels = headingsObject.map(function (headingObject) {
```

Comment:
This seems like it has extra logic. It should simply use the callback if passed in otherwise use the default content. Also, the variable above should probably move in here for perf reasons.

Reply:
I'm not sure I've understood you completly :confused: 
I've pushed a new version, is that better?